runtime/str.go

```diff
@@ -1093,3 +1112,21 @@ func toUpper(b byte) byte {
 	}
 	return b
 }
+
+func xfill(s string, width int, fillchar string) string {
+	l := len(s)
+	if width <= l {
+		return s
+	}
+	buf := bytes.Buffer{}
+	buf.Grow(width)
+	if l > 0 && fillchar == "0" && (s[0] == '-' || s[0] == '+') {
+		buf.WriteByte(s[0])
+		s = s[1:]
+		width--
+	}
+	// TODO: Support or throw fillchar len more than one.
```

Comment:
I think it's worth mentioning in a comment that len(fillchar) == 1 is a precondition. E.g.:

```
// xfill returns s padded with fillchar so that its length is at least width. fillchar must be a single character.
```

Probably also worth mentioning the fillchar == "0" behavior.